go-controller/pkg/node/node.go

```diff
@@ -330,7 +330,22 @@ func deleteServiceConntrackEntries(svc *kapi.Service) {
 		}
 		err := deleteConntrack(svc.Spec.ClusterIP, port.Port, port.Protocol)
 		if err != nil {
-			klog.Errorf("Failed to delete conntrack entry for %s:%d: %v", svc.Spec.ClusterIP, port.Port, err)
+			klog.Errorf("Failed to delete conntrack entry for ClusterIP %s:%d: %v", svc.Spec.ClusterIP, port.Port, err)
+		}
+		// delete the conntrack entries for nodePorts
+		if port.NodePort > 0 {
+			localAddrSet, err := getLocalAddrs()
+			if err != nil {
+				klog.Errorf("Error obtaining local IP address for NodePort on service %s:%d: %v", svc.Spec.ClusterIP, port.Port, err)
+				continue
+			}
+			for _, address := range localAddrSet {
```

Comment:
Also, `getLocalAddrs()` returns every address associated with every unicast interface on the node. I think here it's enough to delete the conntrack entries for the node's IP address. I.e: you can use `getDefaultIfAddr` or `getNetworkInterfaceIPAddresses`

Reply:
it depends on which addresses have the NodePort open, I couldn' t find consistency on this, but we should clean all the addresses that have the NodePort open

Reply:
you are right, this wasn't the right call